modules/datahandling/datahandler.go

```diff
@@ -63,8 +83,41 @@ func (dh DataHandler) Handle(messageType int, message []byte) error {
 	return err
 }
 
-func authenticate(abs abstractRequest) bool {
-	fmt.Println("AUTHENTICATION IS NOT IMPLEMENTED YET")
-	// TODO (non-immediate/required): implement user authentication
-	return true
+type tokenPayload struct {
+	Username     string
+	CreationTime int64
+	Validity     int64
+}
+
+func (tokenPayload) Valid() error {
+	return nil
+}
```

Comment:
Normally, this is where we would check that the token is valid. However, we need access to the abstract request to make sure that the token's Username and the request's senderID is the same. Thus, the logic had to be handled outside of the Valid method, in the authenticate method just below.

This method then became redundant, but still needed an implementation for interface reasons.

I could move the token time checks into the valid method, but that would separate it from the username checks.